src/Mod/Part/BasicShapes/SkipGui.py

```diff
@@ -0,0 +1,250 @@
+# SkipGui.py
+# 2022, by Mark Ganson <TheMarkster>
+# LGPL 2.1 or later
+
+# this file is called from c++ in Mod/Part/Gui/Command.cpp
+# from various functions when these commands are activated:
+#   Part::Loft
+#   Part::Sweep
+#   Part::Extrude
+#   Part::Revolve
+# Then if the user has preselected certain objects and subobjects
+# the Gui can be skipped and the document objects directly created
+
+# If False is returned, then the Gui dialog should be shown
+# Else if True is returned, then the dialog can be skipped
+
+import FreeCAD as App
+import FreeCADGui as Gui
+
+def getActStr(document):
+    return "App.getDocument(\'" + document.Name + "\').ActiveObject"
+
+def getDocStr(document):
+    return "App.getDocument(\'" + document.Name +"\')"
+
+def getObjStr(document, object):
+    return getDocStr(document) + ".getObject(\'" + object.Name + "\')"
+
+def getVisStr(document, objs):
+    ret = ""
+    for obj in objs:
+        ret += getDocStr(document) + ".getObject(\'" + obj.Name + "\').ViewObject.Visibility = False\n"
+    return ret
+
+def getSetPropListStr(document, propName, objList):
+    ret = getActStr(document) + "." + propName + " = ["
+    for obj in objList:
+        ret += getDocStr(document) + ".getObject(\'" + obj.Name + "\'), "
+    ret += "]\n"
+    return ret
+
+def getAddObjStr(document, typeId, name):
+    ret = getDocStr(document)
+    ret += ".addObject(\'" + typeId + "\', \'" + name + "\')\n"
+    return ret
+
+def getSubListStr(names):
+    ret = "["
+    for name in names:
+        ret += "\'" + name + "\', "
+    ret += "]"
+    return ret
+
+
+def makeExtrude():
+
+    selx = Gui.Selection.getSelectionEx()
+    if len(selx) == 0:
+        App.Console.PrintMessage(f"makeExtrude() len(selx) = {len(selx)} (must be > 0 to skip dialog)\n")
+        return False
+
+    selected = [sel.Object for sel in selx if bool(sel.Object.isDerivedFrom("Part::2DObject") or
+                sel.Object.isDerivedFrom("Part::FeaturePython") or
+                bool(hasattr(sel.Object,"Shape") and len(sel.Object.Shape.Vertexes) == 1) or
+                sel.Object.isDerivedFrom("Sketcher::SketchObject")) and
+                sel.HasSubObjects == False]
+    if len(selected) != 1:
+        App.Console.PrintMessage(f"makeExtrude() len(selected) = {len(selected)} (must be 1 to skip dialog)\n")
+        return False #need exactly 1 profile for Extrude
+
+    solid = "True"
+    base = selected[0]
+    if hasattr(base.Shape, "Wire1"):
+        if not base.Shape.Wire1.isClosed():
+            solid = "False"
+    else:
+        if not bool(hasattr(base,"Shape") and len(base.Shape.Vertexes) == 1):
+            return False #no Wire1
+        else:
+            solid = "False"
+
+    spines = [sel for sel in selx if sel.HasSubObjects]
+    if len(spines) != 1:
+        return False #all edges must be from the same object
+
+    spineEdges = spines[0].SubElementNames
+    subNames = [name for name in spineEdges if "Edge" in name]
```

Comment:
Would me more proper to use `SubObjects` then filter using `TypeId` of `isInstance`

Reply:
Perhaps if we were only interested in the subobjects, but we need the subname not the subobject so we can pass the name to doCommand().